### PR TITLE
docs: update future development blurbs

### PR DESCRIPTION
**Pull request by 317brian**
This PR updates/removes promises of future development for features. Mentions of future removal of things remain.

This PR has:

- [x] been self-reviewed.
   - [ ] using the [concurrency checklist](https://github.com/apache/druid/blob/master/dev/code-review/concurrency.md) (Remove this item if the PR doesn't have any relation to concurrency.)
- [ ] added documentation for new or modified features or behaviors.
- [ ] a release note entry in the PR description.
- [ ] added Javadocs for most classes and all non-trivial methods. Linked related entities via Javadoc links.
- [ ] added or updated version, license, or notice information in [licenses.yaml](https://github.com/apache/druid/blob/master/dev/license.md)
- [ ] added comments explaining the "why" and the intent of the code wherever would not be obvious for an unfamiliar reader.
- [ ] added unit tests or modified existing tests to cover new code paths, ensuring the threshold for [code coverage](https://github.com/apache/druid/blob/master/dev/code-review/code-coverage.md) is met.
- [ ] added integration tests.
- [ ] been tested in a test Druid cluster.

### CODE REVIEW

File: docs/design/architecture.md [hunk 1]
```diff
@@ -105,10 +105,9 @@ for reading from external data sources and publishing new Druid segments.
 [**Indexer**](../design/indexer.md) services are an alternative to MiddleManagers and Peons. Instead of
 forking separate JVM processes per-task, the Indexer runs tasks as individual threads within a single JVM process.
 
-The Indexer is designed to be easier to configure and deploy compared to the MiddleManager + Peon system and to better enable resource sharing across tasks. The Indexer is a newer feature and is currently designated [experimental](../development/experimental.md) due to the fact that its memory management system is still under
-development. It will continue to mature in future versions of Druid.
+The Indexer is designed to be easier to configure and deploy compared to the MiddleManager + Peon system and to better enable resource sharing across tasks, which can help streaming ingestion. The Indexer is currently designated [experimental](../development/experimental.md).
 
-Typically, you would deploy either MiddleManagers or Indexers, but not both.
+Typically, you would deploy MiddleManagers, [MiddleManager-less ingestion using Kubernetes](../development/extensions-contrib/k8s-jobs.md), or Indexers.
```
Comment (317brian) [inline docs/design/architecture.md:110]:
```suggestion
Typically, you would deploy one of the following MiddleManagers, [MiddleManager-less ingestion using Kubernetes](../development/extensions-contrib/k8s-jobs.md), or Indexers. You wouldn't more than one of these options.
```

File: docs/design/architecture.md [hunk 2]
```diff
@@ -105,10 +105,9 @@ for reading from external data sources and publishing new Druid segments.
...
 forking separate JVM processes per-task, the Indexer runs tasks as individual threads within a single JVM process.
 
-The Indexer is designed to be easier to configure and deploy compared to the MiddleManager + Peon system and to better enable resource sharing across tasks. The Indexer is a newer feature and is currently designated [experimental](../development/experimental.md) due to the fact that its memory management system is still under
-development. It will continue to mature in future versions of Druid.
+The Indexer is designed to be easier to configure and deploy compared to the MiddleManager + Peon system and to better enable resource sharing across tasks, which can help streaming ingestion. The Indexer is currently designated [experimental](../development/experimental.md).
 
-Typically, you would deploy either MiddleManagers or Indexers, but not both.
+Typically, you would deploy one of the following MiddleManagers, [MiddleManager-less ingestion using Kubernetes](../development/extensions-contrib/k8s-jobs.md), or Indexers. You wouldn't deploy more than one of these options.
```
Comment (vtlim) [inline docs/design/architecture.md:110]: Should this be a list of three options? "deploy one of the following:"

File: docs/querying/datasource.md [hunk 3]
```diff
@@ -431,25 +431,21 @@ and how to detect it.
...
 
-Joins are an area of active development in Druid. The following features are missing today but may appear in
-future versions:
+Joins in Druid have the following limitations:
 
-- Reordering of join operations to get the most performant plan.
-- Preloaded dimension tables that are wider than lookups (i.e. supporting more than a single key and single value).
-- RIGHT OUTER and FULL OUTER joins in the native query engine. Currently, they are partially implemented. Queries run
+- The order of joins is not entirely optimized. Join operations are not reordered to get the most performant plan.
+- Preloaded Dimension tables that are wider than lookups (i.e. supporting more than a single key and single value) are not supported.
```
Comment (vtlim) [inline docs/querying/datasource.md:445]:
```suggestion
- Preloaded dimension tables that are wider than lookups (i.e. supporting more than a single key and single value) are not supported.
```

File: docs/design/architecture.md (hunk 2 above)
Comment (317brian) [reply docs/design/architecture.md:110]:
```suggestion
Typically, you would deploy one of the following: MiddleManagers, [MiddleManager-less ingestion using Kubernetes](../development/extensions-contrib/k8s-jobs.md), or Indexers. You wouldn't deploy more than one of these options.
```

**Review by vtlim — APPROVED** [review]

**Pull request merged**